static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -64,6 +74,50 @@ export class DownloadTemplateComponent implements OnInit {
         this.authService.logout(null);
     }
 
+    storeRating(new_rating) {
+        const component = this;
+        let authorUID:string  =  component.user.uid;
+        component.templateRatingsInfoDatabaseRef.once('value').then(snapshot2 => {
+            const old_rating = snapshot2.child(authorUID).val(); //value of previous rating
+            let hasRated:boolean = old_rating != null;
+            if (hasRated == true) {
+                // author has reviewed before
+                // update values in template directory
+                component.templateDirectoryInfoDatabaseRef.child('/ratingSum').once('value', function(snapshot) {
```

Comment:
Should use a transaction for these types of things (https://firebase.google.com/docs/firestore/manage-data/transactions)